src/core/cwa.ts

```diff
@@ -20,35 +20,47 @@ export default class Cwa {
       process.env.NODE_TLS_REJECT_UNAUTHORIZED = '0'
     }
 
-    // this class is accessible throughout the application via this.$cwa or $cwa in the context
-
-    // For auth - we are using a nuxt module https://dev.auth.nuxtjs.org/
-    // ctx.$auth
-
-    // For axios we use a nuxt module https://axios.nuxtjs.org/ so it is
-    // the same in all the contexts
-    // ctx.$axios
     this.ctx = ctx
 
     this.fetcher = async ({ path, preload }) => {
       const url = `${process.env.baseUrl}${path}`
       consola.debug('Fetching %s', url)
 
       const requestHeaders = preload ? { Preload: preload.join(',') } : {}
-      // While https://github.com/nuxt-community/auth-module/pull/726 is pending, disable the header
-      ctx.$axios.setHeader('Authorization', false)
-      const { data, headers } = await ctx.$axios.get(url, { headers: requestHeaders })
-      this.getMercureHub(headers)
-      return data
+      if (process.server) {
+        let referer = ctx.req.headers.referer
+        if (!referer) {
+          referer = ctx.req.socket.encrypted ? 'https' : 'http' +
+            '://' +
+            ctx.req.host +
+            ctx.route.fullPath
+        }
+        Object.assign(requestHeaders, { referer })
```

Comment:
what's this for?

Reply:
I've changed this to a querystring. When fetching a `componentPosition` there is a feature in the API that looks in the `pageData` for a given route to populate from that. It's for dynamic pages like blog articles where the page structure is the same but components/data can vary on each article (for example). But yes, I'll push progress in a mo.